### PR TITLE
Add development roadmap to docs

### PR DESCRIPTION
**Pull request by leewujung**
This is to address #602.

A couple of questions:
- Where do we want this page to go?
- Where should the data processing levels table go (I think it's too long to be in this page)?

This PR still has a few links to be fixed but the content is ready for review.

Note that right now this is PR to the `dev` branch but we will add this to `stable` once merged.

### CODE REVIEW

**Comment by leewujung** [comment]
@emiliom : I've done a first pass of the roadmap text, see what you think.

I think that the data processing levels draft table should be moved somewhere else, but I am not sure where under the current Getting started/User guide/Help & Reference sections. Suggestions?

File: docs/source/convert.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+.. _convert:
```
Comment (emiliom) [inline docs/source/convert.rst:1]: What's this? Is it a labelled anchor to refer to the page more explicitly? 

FYI, this one shows up as a string in the rendered docs. The one in `roadmap.md` doesn't. I can only that it's b/c  this page is rst and the other one md.

Comment (leewujung) [reply docs/source/convert.rst:1]: This is some remnant of trying to cross-reference .rst, but can be safely removed now that we are in jupyter book! I think we have whole bunch of such cross-ref things needing revision/deletion.

**Comment by emiliom** [comment]
> Where do we want this page to go?

I think it fits more naturally next to the "Contributing to echopype" page. Before or after, but I lean towards having it before. That would be consistent with what's done in the pandas and xarray docs.

> I think that the data processing levels draft table should be moved somewhere else, but I am not sure where under the current Getting started/User guide/Help & Reference sections. Suggestions?

I definitely agree that it should go somewhere else. It's not clear where, and maybe we're not there yet. Product levels is aspirational at this time, and for now it may have more to do with labelling/categorization, attribution and provenance than with significant echopype software features per se.

**Comment by leewujung** [comment]
> > I think that the data processing levels draft table should be moved somewhere else, but I am not sure where under the current Getting started/User guide/Help & Reference sections. Suggestions?
> 
> I definitely agree that it should go somewhere else. It's not clear where, and maybe we're not there yet. Product levels is aspirational at this time, and for now it may have more to do with labelling/categorization, attribution and provenance than with significant echopype software features per se.

**Comment by emiliom** [comment]
@leewujung I've done a substantial (but not complete, yet) rewrite of your roadmap page. But when I tried to push my commit to your PR, I got the permission error:
```
To github.com:leewujung/echopype.git
 ! [remote rejected] leewujung-docs-roadmap -> leewujung-docs-roadmap (permission denied)
error: failed to push some refs to 'github.com:leewujung/echopype.git'
```

I need to hop on a call right now, so to make it visible to you, I've pushed it to my fork:
https://github.com/emiliom/echopype/blob/leewujung-docs-roadmap/docs/source/roadmap.md

**Comment by leewujung** [comment]
That is a bizarre permission error, I don't remember if it takes the same form as the one that has been haunting us from GitLFS, but I know the symptom of that one was exactly that you would not be able to push to forked branches. I can pull in changes in your fork and work from my end. I need to work on the data format restructuring PR first though, so may not get to this today, so please go ahead and continue what you haven't finished.

**Comment by emiliom** [comment]
Will do. I'll keep working on this later this morning.

**Comment by leewujung** [comment]
Did I accidentally merge this some 13 days ago?? Oops-- @emiliom please open another PR!

**Comment by emiliom** [comment]
I was confused yesterday when looking at this! I actually have an updated roadmap page ready, and creating a new PR will be cleanest.

**Pull request merged**